api/src/main/java/org/apache/iceberg/transforms/Hours.java

```diff
@@ -57,15 +58,16 @@ public boolean satisfiesOrderOf(Transform<?, ?> other) {
     }
 
     if (other instanceof Timestamps) {
-      return other == Timestamps.HOUR;
-    } else if (other instanceof Hours
-        || other instanceof Days
-        || other instanceof Months
-        || other instanceof Years) {
-      return true;
+      return ((Timestamps) other).getResultTypeUnit() == ChronoUnit.HOURS
```

Comment:
Can we cast once and store it in a local variable?

Reply:
Fixed in 3b87435b31eac16b86b4118e79cb54fabf367b64